src/offlinelogstorage/dlt_offline_logstorage_behavior.c

```diff
@@ -501,6 +503,7 @@ DLT_STATIC void dlt_logstorage_open_log_output_file(DltLogStorageFilterConfig *c
 #ifdef DLT_LOGSTORAGE_USE_GZIP
         dlt_vlog(LOG_DEBUG, "%s: Opening GZIP log file\n", __func__);
         config->gzlog = gzdopen(config->fd, mode);
+        config->log = file;
```

Comment:
there should be no need to store config->log here. As per my understanding gzclose() will close the passed fd from gzdopen()

Reply:
According to my tests, after gzclose, fclose is also needed. This part of the test is included in #655

Reply:
@michael-methner  
I did a test to remove fclose, and the results are as follows: fclose is necessary, and the currently submitted code is necessary.

add fclose
![add fclose](https://github.com/user-attachments/assets/5a0779a8-704a-46b5-9ce1-a6bed1149411)

delete fclose
![delete_flocse](https://github.com/user-attachments/assets/1205ecb4-e5dd-4d74-a665-f9aa401de10a)